playgame.py

```diff
@@ -5,55 +5,111 @@
 from tools.hanabi_card import HanabiColor
 import sys
 import logging
+from logging.handlers import RotatingFileHandler
+import os
 from tools.hanabi_moves import \
     HanabiDiscardAction, \
     HanabiPlayAction, \
     HanabiColorDiscloseAction, \
     HanabiRankDiscloseAction
 
-logger = logging.getLogger(__name__)
-
 def main(argv):
     args = parse_args()
-    prep_logger(args.log_dir, args.verbose, args.log_stderr)
+    logger = prep_logger(args.log_dir, args.verbose, args.log_stderr, len(args.players))
+    if args.is_tournament:
+        run_tournament(args, logger)
+    else:
+        run_one_game(args, logger)
+
+def generate_pairings(players):
+    res = []
+    for i in range(len(players)):
+        for j in range(i + 1, len(players)):
+            res.append([players[i], players[j]])
+    return res
+
+def run_tournament(args, logger):
+    tournament_scores = dict.fromkeys(args.players, 0)
+    pairings = generate_pairings(args.players)
+    for player1, player2 in pairings:
+        try:
+            game = HanabiGame([player1, player2], args.seed, args.variant)
+            game.play_game(args, logger)
+            score = game.table.score()
+        except:
+            #if a player has messed up, penalize both
+            #could diqualify the failing bot if we can determine who failed
+            logger.warning("{player1} or {player2} failed to complete a game".format(
+                player1 = player1,
+                player2 = player2
+                ))
+            score = -25
+        finally:
+            for handler in logger.handlers:
+                if handler.__class__ is RotatingFileHandler:
+                    handler.doRollover()
```

Comment:
I'm sort of surprised `RotatingFileHandler` doesn't auto-rotate on start-up (my only experience will log rolling is log4j2, which does this). Kind of a bummer to have to manually rotate like this, but doing some sort of custom strategy doesn't seem like the best idea.

Reply:
RotatingFileHandler rotates automatically when given a bytesize or time duration. I'm not including those, since I want the rotation at the end of a game. Can check to see if there's a better option (#25)